### DISCUSSION TITLE
API Improvements

### DISCUSSION THREAD

**Discussion by agardnerIT** [infracost/infracost#1945] [Ideas & Feature Requests]
A few suggestions for the `pricing.api.infracost.io` endpoints described [here](https://www.infracost.io/docs/integrations/infracost_api/):

1. Move form params to headers
2. Allow raw JSON in body
3. Allow JSON output format on `/diff` endpoint

It would be much simpler for users (CI integrations) if the API allowed a plan.json content (the actual JSON) to be passed in the POST body and not as multipart form data.

```
headers = {
  "x-api-key": "abc1234",
  "format": "json",
  "ci-platform": "my-platform"
}

plan = {
  # terraform plan here
}

requests.post(url="https://pricing.api.infracost.io/breakdown", headers=headers, data=plan)
```

**Comment by alikhajeh1** [comment]
Thanks @agardnerIT! I tweaked the description to clarify that this idea is for https://www.infracost.io/docs/integrations/infracost_api/
